plugin/src/main/java/org/opensearch/ml/action/stats/MLStatsNodesTransportAction.java

```diff
@@ -157,29 +157,24 @@ MLStatsNodeResponse createMLStatsNodeResponse(MLStatsNodesRequest mlStatsNodesRe
 
         Map<String, MLModelStats> modelStats = new HashMap<>();
         if (mlStatsInput.includeModelStats()) {
-            CountDownLatch latch = new CountDownLatch(1);
             boolean isSuperAdmin = isSuperAdminUserWrapper(clusterService, client);
-            searchHiddenModels(ActionListener.wrap(hiddenModels -> {
-                for (String modelId : mlStats.getAllModels()) {
-                    if (isSuperAdmin || !hiddenModels.contains(modelId)) {
-                        if (mlStatsInput.retrieveStatsForModel(modelId)) {
-                            Map<ActionName, MLActionStats> actionStatsMap = new HashMap<>();
-                            for (Map.Entry<ActionName, MLActionStats> entry : mlStats.getModelStats(modelId).entrySet()) {
-                                if (mlStatsInput.retrieveStatsForAction(entry.getKey())) {
-                                    actionStatsMap.put(entry.getKey(), entry.getValue());
-                                }
+            Set<String> hiddenModels = mlStatsNodesRequest.getHiddenModelIds() != null
+                ? mlStatsNodesRequest.getHiddenModelIds()
+                : Collections.emptySet();
+            for (String modelId : mlStats.getAllModels()) {
+                if (isSuperAdmin || !hiddenModels.contains(modelId)) {
+                    if (mlStatsInput.retrieveStatsForModel(modelId)) {
+                        Map<ActionName, MLActionStats> actionStatsMap = new HashMap<>();
+                        for (Map.Entry<ActionName, MLActionStats> entry : mlStats.getModelStats(modelId).entrySet()) {
+                            if (mlStatsInput.retrieveStatsForAction(entry.getKey())) {
```

Comment:
We should refactor this 5 layer of nested for and ifs. This is far from elegant. At least we can use stream().filter(...) to avoid some.

Reply:
+1, simplifying this.